05/app.js

```diff
@@ -0,0 +1,46 @@
+document.addEventListener('DOMContentLoaded', init);
+
+function init() {
+  const form = document.querySelector('form');
+  const message = document.querySelector('.messages');
+
+  form.addEventListener('submit', function (e) {
+    e.preventDefault();
+    message.innerHTML = '';
+    const errors = [];
+
+    const firstName = form.elements.firstName.value.trim();
+    const lastName = form.elements.lastName.value.trim();
+    const street = form.elements.street.value.trim();
+    const houseNumber = form.elements.houseNumber.value.trim();
+    const flatNumber = form.elements.flatNumber.value.trim();
+    const zip = form.elements.zip.value.trim();
+    const city = form.elements.city.value.trim();
+    const voivodeship = form.elements.voivodeship.value.trim();
+
+    if (firstName === '') errors.push('Imię jest wymagane.');
```

Comment:
Bardzo dobrze sobie poradziłaś z tą walidacja!
Może spróbujesz zadania dodatkowego tj. spróbuj go wykonać w ten sposób:
https://devmentor.pl/b/walidacja-formularza-w-javascript

Będzie to bardziej uniwersalne i do wykorzystania w przyszłości.